### ISSUE TITLE
Support modifying `Allow forking` option for resource_github_repository

### ISSUE THREAD

**Issue by ganhuang** [integrations/terraform-provider-github#353]
Currently the the option `Allow forking` is not supported in the resource resource_github_repository. But it could be done via UI.

#### Affected Resource(s)
Please list the resources as a list, for example:
- resource_github_repository

**Comment by benj-fletch** [comment]
@ganhuang I don't believe that this is possible to be configured via the [Github API](https://developer.github.com/v3/repos/) for repositories? Do you know another way that this could be supported?

**Comment by deppy** [comment]
@benj-fletch https://developer.github.com/v3/repos/forks/#create-a-fork

**Comment by kfcampbell** [comment]
We could use Terraform to create a fork, certainly, but read/update/delete operations still aren't supported as far as I can see.

**Comment by timhirsh** [comment]
It looks like the ask in this issue was misunderstood (and the issue prematurely closed due to staleness).  The `allow_forking` option is part of the "update repository" api here: https://docs.github.com/en/free-pro-team@latest/rest/repos/repos?apiVersion=2022-11-28#update-a-repository

@kfcampbell, https://github.com/integrations/terraform-provider-github/pull/1033 appears to have implemented this but it was reverted shortly after.

**Comment by kfcampbell** [comment]
@timhirsh you're correct; I'll reopen and pin this issue.

**Comment by deiga** [comment]
Resolved by https://github.com/integrations/terraform-provider-github/pull/3056
